src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -509,10 +509,9 @@ public Participation findOneByExerciseIdAndStudentLogin(Long exerciseId, String
                             // we are only interested in results with completion date and with score
                             continue;
                         }
-                        if (participation.getExercise() instanceof QuizExercise) {
-                            // in quizzes we take all rated results, because we only have one! (independent of later checks)
-                        }
-                        else if (participation.getExercise().getDueDate() != null) {
+
+                        // in quizzes we take all rated results, because we only have one! (independent of later checks)
+                        if (!(participation.getExercise() instanceof QuizExercise) && participation.getExercise().getDueDate() != null) {
```

Comment:
please revert this change, we intentionally wrote this code like this

Reply:
Done